src/OrleansRuntime/Catalog/DefaultGrainActivator.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Globalization;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Orleans.Runtime
+{
+    /// <summary>
+    /// <see cref="IGrainActivator"/> that uses type activation to create grains.
+    /// </summary>
+    public class DefaultGrainActivator : IGrainActivator
+    {
+        private readonly Func<Type, ObjectFactory> createFactory = type => ActivatorUtilities.CreateFactory(type, Type.EmptyTypes);
```

Comment:
Isn't this all that is needed?  A hook to create the grain factory?

Func<Type, object[] parameters, ObjectFactory>?

Get this one thing from DI, and nothing else needs to change, right?